torch_em/data/datasets/light_microscopy/omnipose.py

```diff
@@ -0,0 +1,145 @@
+
+    zip_path = os.path.join(path, "datasets.zip")
+    util.download_source(path=zip_path, url=URL, download=download, checksum=CHECKSUM)
+    util.unzip(zip_path=zip_path, dst=data_dir)
+
+    return data_dir
+
+
+def _get_omnipose_paths(path, split, data_choice, download):
+    data_dir = get_omnipose_data(path=path, download=download)
+    data_dir = os.path.join(data_dir, data_choice)
+
+    if split not in ["train", "test"]:
+        raise ValueError(f"'{split}' is not a valid split.")
+
+    if data_choice not in ["bact_fluor", "bact_phase", "worm", "worm_high_res"]:
+        raise ValueError(f"'{data_choice}' is not a valid choice of data.")
+
+    if data_choice.startswith("bact"):
+        base_dir = os.path.join(data_dir, f"{split}_sorted", "*")
+        gt_paths = glob(os.path.join(base_dir, "*_masks.tif"))
+        image_paths = glob(os.path.join(base_dir, "*.tif"))
+        for _path in image_paths.copy():
+            if _path.endswith("_masks.tif") or _path.endswith("_flows.tif"):
+                image_paths.remove(_path)
+
+    else:
+        base_dir = os.path.join(data_dir, split)
+        gt_paths = glob(os.path.join(base_dir, "*_masks.*"))
+        image_paths = glob(os.path.join(base_dir, "*"))
+        for _path in image_paths.copy():
+            if _path.endswith("_masks.tif") or _path.endswith("_masks.png") or _path.endswith("_flows.tif"):
+                image_paths.remove(_path)
+
+    return natsorted(image_paths), natsorted(gt_paths)
+
+
+def get_omnipose_dataset(
+    path: Union[os.PathLike, str],
+    patch_shape: Tuple[int, int],
+    split: Literal["train", "test"],
+    data_choice: Literal["bact_fluor", "bact_phase", "worm", "worm_high_res"],
+    download: bool = False,
+    **kwargs
+):
+    """Get the OmniPose dataset for segmenting bacteria and worms in microscopy images.
+
+    Args:
+        path: Filepathg to a folder where the downloaded data will be saved.
+        patch_shape: The patch shape to use for training.
+        split: The data split to use. Either 'train' or 'test'.
+        download: Whether to download the data if it is not present.
+        data_choice: The choice of specific data. Either 'bact_fluor', 'bact_phase', 'worm' or 'worm_high_res'.
+        kwargs: Additional keyword arguments for `torch_em.default_segmentation_dataset`.
+
+    Returns:
+        The segmentation dataset.
+    """
+    image_paths, gt_paths = _get_omnipose_paths(path, split, data_choice, download)
+
+    dataset = torch_em.default_segmentation_dataset(
+        raw_paths=image_paths,
+        raw_key=None,
+        label_paths=gt_paths,
+        label_key=None,
+        is_seg_dataset=False,
+        patch_shape=patch_shape,
+        **kwargs
+    )
+    return dataset
+
+
+def get_omnipose_loader(
+    path: Union[os.PathLike, str],
+    patch_shape: Tuple[int, int],
+    batch_size: int,
+    split: Literal["train", "test"],
+    data_choice: Literal["bact_fluor", "bact_phase", "worm", "worm_high_res"],
```

Comment:
Would it make sense to add `None` as (the default?) option and then load all the different data types?
(I am not sure, we don't have to.)